### ISSUE TITLE
How to fix “No semisync settings” 

### ISSUE THREAD

**Issue by tanglu86** [signal18/replication-manager#539]
replication-manager  ： 2.2.16
Mysql ： 8.0.35
replicaton mode :GTID+ SEMI(AFTER_SYNC )

The MASTER node and SLAVE node both have SEMI replication turned on, but RM interface keeps showing "No semisync settings on slave" and "No semisync settings on master".

![360截图20240221122404878](https://github.com/signal18/replication-manager/assets/38269740/c30c8bd0-b5ef-4b36-9702-ed45a2b02df8)

![2](https://github.com/signal18/replication-manager/assets/38269740/642b5e2a-6cbb-4ffb-8602-516f899ba4d9)

**Comment by svaroqui** [comment]
Hi looks like we are looking at status and i guess the naming hav change in mysql 

```
func (server *ServerMonitor) IsSemiSyncMaster() bool {
	return server.Status["RPL_SEMI_SYNC_MASTER_STATUS"] == "ON" || server.Status["RPL_SEMI_SYNC_SOURCE_STATUS"] == "ON"
}

func (server *ServerMonitor) IsSemiSyncReplica() bool {
	return server.Status["RPL_SEMI_SYNC_SLAVE_STATUS"] == "ON" || server.Status["RPL_SEMI_SYNC_REPLICA_STATUS"] == "ON"
}
```

**Comment by svaroqui** [comment]
I guess only MariaDB can rollback XA uncommitted TRX after a crash so that the old leader can rejoin the new leader without backup, if you know more than me on this please let me know , as i know facebook that rely on mysql have also produce some patch to do the same, but do not know if they came to the trunk ?

**Comment by svaroqui** [comment]
Nota : rename on 8.0
INSTALL PLUGIN rpl_semi_sync_master SONAME 'semisync_master.so';

Or from MySQL 8.0.26: e also need to fixe this :
INSTALL PLUGIN rpl_semi_sync_source SONAME 'semisync_source.so';

show variables like '%sem%'
    -> ;
+---------------------------------------------+------------+
| Variable_name                               | Value      |
+---------------------------------------------+------------+
| rpl_semi_sync_source_enabled                | OFF        |
| rpl_semi_sync_source_timeout                | 10000      |
| rpl_semi_sync_source_trace_level            | 32         |
| rpl_semi_sync_source_wait_for_replica_count | 1          |
| rpl_semi_sync_source_wait_no_replica        | ON         |
| rpl_semi_sync_source_wait_point             | AFTER_SYNC |
+---------------------------------------------+------------+

MySQL liek to give us unecessary work :)
 show status like '%sem%'
    -> ;
+--------------------------------------------+-------+
| Variable_name                              | Value |
+--------------------------------------------+-------+
| Rpl_semi_sync_source_clients               | 0     |
| Rpl_semi_sync_source_net_avg_wait_time     | 0     |
| Rpl_semi_sync_source_net_wait_time         | 0     |
| Rpl_semi_sync_source_net_waits             | 0     |
| Rpl_semi_sync_source_no_times              | 0     |
| Rpl_semi_sync_source_no_tx                 | 0     |
| Rpl_semi_sync_source_status                | OFF   |
| Rpl_semi_sync_source_timefunc_failures     | 0     |
| Rpl_semi_sync_source_tx_avg_wait_time      | 0     |
| Rpl_semi_sync_source_tx_wait_time          | 0     |
| Rpl_semi_sync_source_tx_waits              | 0     |
| Rpl_semi_sync_source_wait_pos_backtraverse | 0     |
| Rpl_semi_sync_source_wait_sessions         | 0     |
| Rpl_semi_sync_source_yes_tx                | 0     |
+--------------------------------------------+-------+
14 rows in set (0,015 sec)

**Comment by tanglu86** [comment]
> Hi looks like we are looking at status and i guess the naming hav change in mysql
> 
> ```
> func (server *ServerMonitor) IsSemiSyncMaster() bool {
> 	return server.Status["RPL_SEMI_SYNC_MASTER_STATUS"] == "ON" || server.Status["RPL_SEMI_SYNC_SOURCE_STATUS"] == "ON"
> }
> 
> func (server *ServerMonitor) IsSemiSyncReplica() bool {
> 	return server.Status["RPL_SEMI_SYNC_SLAVE_STATUS"] == "ON" || server.Status["RPL_SEMI_SYNC_REPLICA_STATUS"] == "ON"
> }
> ```

The name of the repl plugin has indeed changed.
Before 8.0: rpl_semi_sync_master, rpl_semi_sync_slave
After 8.0: rpl_semi_sync_source, rpl_semi_sync_replica
Has this issue been fixed in RM V2.3.15?

**Comment by svaroqui** [comment]
I'm not able to reproduce when i set rpl_semi_sync_source_enabled=1; on all servers the warning is removed
